src/agent/src/image_rpc.rs

```diff
@@ -37,73 +38,58 @@ impl ImageService {
+        let oci_path = tmp_cid_path.join(IMAGE_OCI);
         let target_path_oci = format!("oci://{}", oci_path.to_string_lossy());
 
-        fs::create_dir_all(&manifest_path)?;
         fs::create_dir_all(&oci_path)?;
 
         info!(sl!(), "Attempting to pull image {}...", &source_image);
 
         let mut pull_command = Command::new(SKOPEO_PATH);
         pull_command
-            // TODO: need to create a proper policy
-            .arg("--insecure-policy")
             .arg("copy")
             .arg(source_image)
-            .arg(&target_path_manifest);
+            .arg(&target_path_oci)
+            .arg("--remove-signatures");
 
+        // If source credentials were passed (so not using an anonymous registry), pass them through
         if let Some(source_creds) = source_creds {
             pull_command.arg("--src-creds").arg(source_creds);
         }
 
-        let status: ExitStatus = pull_command.status()?;
-        ensure!(
-            status.success(),
-            "failed to copy image manifest: {:?}",
-            status,
-        );
-
-        // Copy image from one local file-system to another
-        // Resulting image is still stored in manifest format, but no longer includes the signature
-        // The image with a signature can then be unpacked into a bundle
-        let status: ExitStatus = Command::new(SKOPEO_PATH)
-            .arg("--insecure-policy")
-            .arg("copy")
-            .arg(&target_path_manifest)
-            .arg(&target_path_oci)
-            .arg("--remove-signatures")
-            .status()?;
+        // If a policy_path provided, use it, otherwise fall back to allow all image registries
+        if let Some(policy_path) = policy_path {
+            pull_command.arg("--policy").arg(policy_path);
+        } else {
+            pull_command.arg("--insecure-policy");
+        }
 
-        ensure!(status.success(), "failed to copy image oci: {:?}", status);
+        let status: ExitStatus = pull_command.status()?;
 
-        // To save space delete the manifest.
-        // TODO LATER - when verify image is added, this will need moving the end of that, if required
-        fs::remove_dir_all(&manifest_path)?;
+        if !status.success() {
+            fs::remove_dir_all(&tmp_cid_path)?;
+            return Err(anyhow!(format!("failed to pull image: {:?}", status)));
+        }
         Ok(())
     }
 
     fn unpack_image(cid: &str) -> Result<()> {
-        let source_path_oci = Self::build_oci_path(cid);
-        let target_path_bundle = format!("{}{}{}", CONTAINER_BASE, "/", cid);
+        let tmp_cid_path = Path::new("/tmp/").join(cid);
+        let source_path_oci = tmp_cid_path.join(IMAGE_OCI);
 
-        info!(sl!(), "unpack image"; "cid" => cid, "target_bundle_path" => &target_path_bundle);
+        let target_path_bundle = Path::new(CONTAINER_BASE).join(cid);
+
+        info!(sl!(), "cid is {:?}", cid);
+        info!(sl!(), "target_path_bundle is {:?}", target_path_bundle);
```

Comment:
Please combine these two log calls into one. We try to minimise log calls but also collate all relevant fields into a single call where possible to make it easier to manipulate. See:

- https://github.com/kata-containers/tests/tree/main/cmd/log-parser

Reply:
Sorry that was my mistake, just a bad merge. Fixed now

Reply:
So I'm confused. Now I've reverted to the original logging I'm getting errors in compilation:
```
error[E0277]: the trait bound `PathBuf: slog::Value` is not satisfied
    --> src/image_rpc.rs:103:9
     |
103  |         info!(sl!(), "unpack image"; "cid" => cid, "target_bundle_path" => &target_path_bundle);
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `slog::Value` is not implemented for `PathBuf`
     | 
```

Reply:
Ah, mystery solved, I've changed the type of `target_path_bundle` to be a Path, so need to get that working with the logging!

Reply:
I've switched it back to a similar log to before, but on a single line.